tests/kernel/device/src/main.c

```diff
@@ -202,6 +202,46 @@ static void test_build_suspend_device_list(void)
 	zassert_false((devcount == 0), NULL);
 }
 
+/**
+ * @brief Test APIs to enable and disable automatic idle power management
+ *
+ * @details Test the API enable and disable, cause we do not implement our PM
+ * API here, it will use the default function to handle power status. So when
+ * we try to get power state by device_get_power_state(), it will default
+ * return power state zero. And we check it.
+ *
+ * @ingroup kernel_device_tests
+ */
+static void test_enable_and_disable_automatic_idle_pm(void)
+{
+	const struct device *dev;
+	int ret;
+	unsigned int device_power_state = 0;
+
+	dev = device_get_binding(DUMMY_PORT_2);
+	zassert_false((dev == NULL), NULL);
+
+	/* check its status at first */
+	/* for cases that cannot run IDLE power, we skip it now */
+	ret = device_get_power_state(dev, &device_power_state);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
```

Comment:
Standard practice for this flow is:

    TC_PRINT("Power management not supported on device");
    ztest_test_skip();
    return;

Reply:
@pabigot Thanks for your comment, I modified the flow of this already.